src/sync/installation.ts

```diff
@@ -118,7 +119,10 @@ export const updateJobStatus = async (
 		scheduleNextTask(0);
 		// no more data (last page was processed of this job type)
 	} else if (!(await getNextTask(subscription, targetTasks))) {
-		await subscription.update({ syncStatus: SyncStatus.COMPLETE });
+		const newBackFillSinceDate = await booleanFlag(BooleanFlags.USE_BACKFILL_ALGORITHM_INCREMENTAL, jiraHost)
+			? calcNewBackfillSinceDate(subscription.backfillSince, data.commitsFromDate, data.syncType, logger)
+			: subscription.backfillSince;
+		await subscription.update({ syncStatus: SyncStatus.COMPLETE, backfillSince: newBackFillSinceDate  });
```

Comment:
I think we need to update this when syncing of commits has been completed instead of whole sync process done. Can't we fetch `newBackFillSinceDate` from message rather than computing it again?

Reply:
Hmm, good questions. I thought the `backfillSinceDate` don't just mean `commits`, but also other tasks as well. Hence I update it only on all job success. Let me ask @Ryan-Yuanqing-Jiang .

`Can't we fetch newBackFillSinceDate` <-- the date in the msg body is the `requested` date. But for existing customers, requested date will be more recent, so the final bakfillSinceDate in DB should still be `undefined`.

----------------

But, **WAIT** !! I think you found a bug!!
For the first time new customers, the existing `backfillSinceDate` is blank, and it should be updated to the requested date. Let me think about it.

Reply:
> But, **WAIT** !! I think you found a bug!! For the first time new customers, the existing `backfillSinceDate` is blank, and it should be updated to the requested date. Let me think about it.

Fixed.